File: lab/timekeeping.py
import time

sum = 0; add = 1

start = time.time()
iterations = 1000*1000*100
for i in range(iterations):
    sum += add
    add /= 2.0

end = time.time()
print("Python for Time measured: {} seconds".format(end - start))

i=0
start = time.time()
iterations = 1000*1000*100
while i < iterations:
    sum += add
    add /= 2.0
    i+=1

end = time.time()
print("Python while Time measured: {} seconds".format(end - start))

File: lab/timekeeping.cpp

#include <stdio.h>
#include <chrono>

int main () {
    double sum(0), add(1);
    auto begin = std::chrono::high_resolution_clock::now();
    
    int iterations = 1000*1000*100;
    for (int i=0; i<iterations; i++) {
        sum += add;
        add /= 2.0;
    }
    
    auto end = std::chrono::high_resolution_clock::now();
    auto elapsed = std::chrono::duration_cast<std::chrono::nanoseconds>(end - begin);
    printf("Result: %.20f\n", sum);
    printf("C++ Time measured: %.3f seconds.\n", elapsed.count() * 1e-9);
    
    return 0;
}